application/classes/Ushahidi/Repository/Contact.php

```diff
@@ -143,6 +144,29 @@ public function isInTargetedSurvey($contact_id)
         return false;
     }
 
+	/**
+	 * @param $contact_id
+	 * @return FALSE or a post_id to reference in the message
+	 */
+    public function hasPostOutsideOfTargetedSurvey($contact_id)
+	{
+		$query_posts = DB::select(DB::expr('DISTINCT (messages.post_id) as post_id'))
+			->from('messages')
+			->where("messages.post_id", 'NOT IN',
+				DB::query
+				(
+					Database::SELECT, 'select targeted_survey_state.post_id from targeted_survey_state where contact_id = :contact'
+				)
+				->bind(':contact', $contact_id))
+			->where(DB::expr('messages.contact_id'), '=', $contact_id);
+
+		$post_ids_in_messages = $query_posts->execute($this->db)->as_array(null, 'post_id');
+		if (count($post_ids_in_messages) >= 1) {
```

Comment:
@kinstelli problem: what if the contact already has multiple posts? obviously this will happen because this limitation is new . I'm just getting one post id but that's not ideal
@rjmackay

Reply:
I agree it's an issue, but I can't think of the best way to solve it yet.

Reply:
I keep thinking that for each incoming message, the flow should be like this:
- if this contact in an active targeted survey, then attach that message to the post for that active survey
- if they're not in an active survey, then we should just find the _latest_ post

On the client, if the user is looking at a targeted survey post, then we could filter out any messages that aren't related to that particular post_id.
If the user is _not_ looking at a targeted survey, then maybe we can include every message for that user _except_ for posts in any targeted surveys.

I might be forgetting something, but this seems like it might account for the situations we've discussed.